packages/near-membrane-base/src/connector.ts

```diff
@@ -8,9 +8,101 @@ const marshallSourceTextInStrictMode = `
+                let stackTrace = '';
+                try {
+                    stackTrace = ReflectApply(ErrorProtoToString, error, []);
+                    // eslint-disable-next-line no-empty
+                } catch {
+                    stackTrace = '<error>';
+                }
+                let consecutive = false;
+                for (let i = 0, { length } = callSites; i < length; i += 1) {
+                    const callSite = callSites[i];
+                    const funcName = ReflectApply(CallSiteProtoGetFunctionName, callSite, []);
+                    let isMarked = false;
+                    if (
+                        typeof funcName === 'string' &&
+                        funcName !== 'eval' &&
+                        ReflectApply(StringProtoEndsWith, funcName, [LOCKER_IDENTIFIER_MARKER])
+                    ) {
+                        isMarked = true;
+                    }
+                    if (!isMarked) {
+                        const evalOrigin = ReflectApply(CallSiteProtoGetEvalOrigin, callSite, []);
+                        if (
+                            typeof evalOrigin === 'string' &&
+                            ReflectApply(StringProtoIncludes, evalOrigin, [
+                                LOCKER_IDENTIFIER_MARKER,
+                            ])
+                        ) {
+                            isMarked = true;
+                        }
+                    }
+                    // Only write a single LWS entry per consecutive LWS stacks.
+                    if (isMarked) {
+                        if (!consecutive) {
+                            consecutive = true;
+                            stackTrace += '\n    at LWS';
+                        }
+                        continue;
+                    } else {
+                        consecutive = false;
+                    }
+                    try {
+                        stackTrace += `\n    at ${ReflectApply(
+                            CallSiteProtoToString,
+                            callSite,
+                            []
+                        )}`;
+                        // eslint-disable-next-line no-empty
+                    } catch {}
+                }
+                return stackTrace;
+            };
+            // Make Error.prepareStackTrace non-configurable and non-writable.
+            Reflect.defineProperty(Error, 'prepareStackTrace', {
+                // @ts-ignore: TS doesn't like __proto__ on property descriptors.
+                __proto__: null,
+                enumerable: true,
+                // Error.prepareStackTrace cannot be a bound or proxy wrapped
+                // function, so to obscure its source we wrap the call to
+                // formatStackTrace().
+                value: function prepareStackTrace(error: Error, callSites: NodeJS.CallSite[]) {
+                    return formatStackTrace(error, callSites);
+                },
+            });
+        }
```

Comment:
How can we test this?

Reply:
We can test it in the Locker PR: https://github.com/salesforce/locker/pull/582

Reply:
Right—great!